narwhal/worker/src/batch_maker.rs

```diff
@@ -267,7 +267,11 @@ impl BatchMaker {
         let store = self.store.clone();
         let worker_id = self.id;
         let tx_digest = self.tx_digest.clone();
-        let metadata = batch.metadata.clone();
+        let mut metadata = batch.metadata.clone();
```

Comment:
I agree it may make more sense to use batch seal time as its creation time. In future we can calculate the average per-txn delay before a batch is sealed, to account for that.

But it seems better to update batch.metadata to the seal time, instead of only sending the seal time to proposer?